packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -450,6 +452,11 @@ export default class CodeGen {
             }
         }
 
+        // TODO [#3370]: remove experimental template expression flag
+        if (this.state.config.experimentalComplexExpressions) {
+            return bindComplexExpression(expression as ComplexExpression, this);
+        }
```

Comment:
Could this be hidden behind a facade? 
In theory, we should be able to treat existing expressions in a similar fashion as complex expressions. With complex expression binding being a more generic implementation.

Reply:
The goal with this initial PR was to ensure zero observable changes. If template expressions move forward, these two implementations should be consolidated. Forking the behavior here accomplishes two things:

1. It makes it very simple to strip out the complex expression support, if we decide not to take the feature to GA.
2. It reduces the risk that some nuance in the binding logic would introduce an observable change when `experimentalComplexExpressions` is `false`.